x/incentives/keeper/distribute_test.go

```diff
@@ -868,3 +871,199 @@ func (s *KeeperTestSuite) TestGetPoolFromGaugeId() {
+// 		- nothing distributes as non-perpetual gauges with 2 epochs have ended and perpetual gauges have not been reloaded
+//		- nothing should change in terms of incentive records
+func (s *KeeperTestSuite) TestFunctionalInternalExternalCLGauge() {
+	// 1. Initialize variables
+	s.SetupTest()
+	const (
+		defaultExternalGaugeValue int64 = 1_000_000
+		defaultInternalGaugeValue int64 = 750_000
+		numEpochsPaidOverGaugeTwo int64 = 2
+	)
+	var (
+		requiredBalances         = sdk.NewCoins(sdk.NewCoin("eth", sdk.NewInt(defaultExternalGaugeValue)), sdk.NewCoin("usdc", sdk.NewInt(defaultExternalGaugeValue)))
+		internalGaugeCoins       = sdk.NewCoins(sdk.NewCoin("stake", sdk.NewInt(defaultInternalGaugeValue)))                                                                                                               // distributed full sum at epoch
+		externalGaugeCoins       = sdk.NewCoins(sdk.NewCoin("eth", sdk.NewInt(defaultExternalGaugeValue)), sdk.NewCoin("usdc", sdk.NewInt(defaultExternalGaugeValue)))                                                     // distributed full sum at epoch
+		halfOfExternalGaugeCoins = sdk.NewCoins(sdk.NewCoin("eth", sdk.NewInt(defaultExternalGaugeValue/numEpochsPaidOverGaugeTwo)), sdk.NewCoin("usdc", sdk.NewInt(defaultExternalGaugeValue/numEpochsPaidOverGaugeTwo))) // distributed at each epoch for non-perp gauge with numEpoch = 2
+
+		internalGaugeDecCoins       = osmoutils.ConvertCoinsToDecCoins(internalGaugeCoins)
+		externalGaugeDecCoins       = osmoutils.ConvertCoinsToDecCoins(externalGaugeCoins)
+		halfOfExternalGaugeDecCoins = osmoutils.ConvertCoinsToDecCoins(halfOfExternalGaugeCoins)
+	)
+
+	s.FundAcc(s.TestAccs[1], requiredBalances)
+	s.FundAcc(s.TestAccs[2], requiredBalances)
+	s.FundModuleAcc(incentivetypes.ModuleName, requiredBalances)
+
+	epochInfo := s.App.IncentivesKeeper.GetEpochInfo(s.Ctx)
+
+	// 2. Setup CL pool and gauge (gauge automatically gets created at the end of CL pool creation).
+	clPoolId1 := s.PrepareConcentratedPool() // creates internal no-lock gauge id = 1
+
+	// check if the gauge is created
+	clPoolInternalGaugeId, err := s.App.PoolIncentivesKeeper.GetPoolGaugeId(s.Ctx, clPoolId1.GetId(), epochInfo.Duration)
+	s.Require().NoError(err)
+
+	clPoolId2 := s.PrepareConcentratedPool() // creates internal no-lock gauge id = 2
+
+	// check if the gauge is created
+	_, err = s.App.PoolIncentivesKeeper.GetPoolGaugeId(s.Ctx, clPoolId2.GetId(), epochInfo.Duration)
+	s.Require().NoError(err)
+
+	// 3. Create external no-lock gauges for CL pools
+	clPoolExternalGaugeIdPool1 := s.CreateNoLockExternalGauges(clPoolId1.GetId(), externalGaugeCoins, s.TestAccs[1], uint64(1))
+	clPoolExternalGaugeIdPool2 := s.CreateNoLockExternalGauges(clPoolId2.GetId(), externalGaugeCoins, s.TestAccs[2], uint64(numEpochsPaidOverGaugeTwo))
+
+	// 4. Create Distribution records to incentivize internal CL no-lock gauges
+	s.IncentivizeInternalGauge([]uint64{clPoolId1.GetId(), clPoolId2.GetId()}, epochInfo.Duration, false)
```

Comment:
This is what I mean. It seems to incentivize both pools. Am I missing anything?

Reply:
yea both pools get internal incentives, later we check that both pool internal incentives have been distributed. This happens at Epoch 2